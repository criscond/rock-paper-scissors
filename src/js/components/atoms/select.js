import { createElement } from '../services';

const select = ({ id, ariaLabel, items, selectedItemId, onChange }) => {
  const element = createElement({
    htmlTag: 'select',
    id,
    ariaLabel,
  });

  items.map((item) => {
    const { id, name } = item;
    var option = createElement({ htmlTag: 'option', ariaLabel: name });
    option.value = id;
    option.text = name;
    option.selected = false;
    if (selectedItemId === id) {
      option.selected = true;
    }
    element.appendChild(option);
  });

  element.addEventListener('change', () => onChange(element.value));

  return element;
}

export default select;
