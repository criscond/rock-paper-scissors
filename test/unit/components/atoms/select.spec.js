import select from '../../../../src/js/components/atoms/select';
describe('Select', () => {
  const onChangeMock = jest.fn();

  const items = [{
    id: 1,
    name: 'Option One'
  }, {
    id: 2,
    name: 'Option Two',
  }, {
    id: 3,
    name: 'Option three'
  }];

  const data = {
    id: 'selectId',
    ariaLabel: 'This is a select',
    items,
    selectedItemId: 2,
    onChange: onChangeMock,
  }

  it('should render a select', () => {
    const element = select(data);

    expect(element).toMatchSnapshot();
  });

  it('should dispatch a change event', () => {
    const element = select(data);
    const documentClickEvent = new Event('change');
    element.dispatchEvent(documentClickEvent);


    expect(onChangeMock).toHaveBeenCalled();
  });
});
