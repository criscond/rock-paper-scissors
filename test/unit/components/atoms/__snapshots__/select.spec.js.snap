// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render a select 1`] = `
<select
  id="selectId"
>
  <option
    value="1"
  >
    Option One
  </option>
  <option
    value="2"
  >
    Option Two
  </option>
  <option
    value="3"
  >
    Option three
  </option>
</select>
`;
